index.html

```diff
@@ -196,10 +197,59 @@
                     },
                     series: series
                 });
+
+                // range annotitation
+                if (window.location.hash.length > 1) {
+                    // #conn-id PLUS start1 PLUS last1 PLUS label1 SEMICOLON start2 ...
+                    // #8100-6+1+3+foo;8200-5+1+7+bar;...
+                    var bands = window.location.hash.substring(1).split(';');
+                    bands.forEach(function (band) {
+                        var parts = band.split('+');
+                        if (parts.length != 4) return;
+                        if (parts[0] !== connid) return;
+                        parts[1] = parseInt(parts[1], 10);
+                        parts[2] = parseInt(parts[2], 10);
+                        graph.xAxis[0].addPlotBand({
+                            from: parts[1],
+                            to: parts[2],
+                            color: 'green',
+                            label: {
+                                text: parts[3]
+                            }
+                        });
+                    });
+                }
+                graphs[connid][reportname] = graph;
             }
         }
     }
+    onHashChange();
+}
+
+function onHashChange() {
+    var bands = window.location.hash.substring(1).split(';');
+    bands.forEach(function (band) {
```

Comment:
A lot of this is duplicated above - should be simple enough to break into a helper. Get and loop bands, and add plot band - optionally provide a graph, otherwise band will be plotted on all graphs.